doc/tutorial/intro-to-autoinstall.rst

```diff
@@ -0,0 +1,239 @@
+means we need a :code:`#cloud-config` header. The autoinstall directives are
+placed under a top level :code:`autoinstall` key, like so:
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+      version: 1
+      ....
+
+Autoinstall on the install media
+--------------------------------
+
+Another option for supplying autoinstall to the Ubuntu installer is to place a
+file named :code:`autoinstall.yaml` on the install media itself.
+
+There are two potential locations for the :code:`autoinstall.yaml` file:
+ * At the root of the "cdrom".  When writing the installation ISO to a USB
+   Flash Drive, this can be done by copying the :code:`autoinstall.yaml` to the
+   partition containing the contents of the ISO - i.e., in the same directory
+   containing the :code:`casper` directory.
+ * On the rootfs of the installation system - this option will typically
+   require modifying the installation ISO and is not suggested, but is
+   supported.
+
+Directly specifying autoinstall as a :code:`autoinstall.yaml` file does not
+require a :code:`#cloud-config` header, and does not use a top level
+:code:`autoinstall` key.  The autoinstall directives are placed at the top
+level.
+
+.. code-block:: yaml
+
+    version: 1
+    ....
+
+
+Cloud-init and Autoinstall interaction
+======================================
+
+While cloud-init may assist in providing the autoinstall configuration to the
+Ubuntu installer, cloud-init itself is not processing the autoinstall.
+
+If cloud-init directives are intended to modify the ephemeral system, they
+must appear at the top level of the cloud-config.  If instead
+cloud-init directives are intended to modify the system being installed, they
+must appear under a :code:`user-data` section in :code:`autoinstall`.
+
+.. code-block:: yaml
+
+    #cloud-config
+    # cloud-init directives may optionally be specified here.
+    # These directives affect the ephemeral system performing the install.
+
+    autoinstall:
+      # autoinstall directives must be specified here, not directly at the top
+      # level.  These directives are processed by the Ubuntu Installer, and
+      # configure the target system to be installed.
+
+      user-data:
+        # cloud-init directives may also be optionally be specified here.
+        # These directives also affect the target system to be installed,
+        # and are processed on first boot.
+
+
+Zero-touch deployment with autoinstall
+======================================
+
+The Ubuntu Installer contains a safeguard, intended to prevent USB Flash Drives
+with an :code:`autoinstall.yaml` file from wiping out the wrong system.
+
+During autoinstall, a prompt will be shown to confirm that the install really
+should proceed and start making modifications to the target system. ::
+
+    start: subiquity/Meta/status_GET
+    Confirmation is required to continue.
+    Add 'autoinstall' to your kernel command line to avoid this
+
+
+    Continue with autoinstall? (yes|no)
```

Comment:
Nothing to do here but I wish we'd stop the install if the user types "no". Right now, it keeps asking the same question again until you type "yes" (at least in dry-run)